### PR TITLE
Move generic iteration function to top level module

### PR DESCRIPTION
**Pull request by juanchopanza**
Function is called neurom.iter_neurites. Code in neurom.core.
Functionality extended to segment functions.

### CODE REVIEW

File: neurom/core/__init__.py [hunk 1]
```diff
@@ -27,3 +27,26 @@
...
+        Get the lengths of sections in a neuron and a population
+
+        >>> from neurom import sections as sec
+        >>> neuron_lengths = [l for l in iter_neurites(nrn, sec.length)]
+        >>> population_lengths = [l for l in iter_neurites(pop, sec.length)]
+        >>> neurite = nrn.neurites[0]
+        >>> tree_lengths = [l for l in iter_neurites(neurite, sec.length)]
+
+    '''
+    #  TODO: optimize case of single neurite and move code to neurom.core.tree
```
Comment (mgeplf) [inline neurom/core/__init__.py]: this todo still relevant?

File: neurom/tests/test_segments.py [hunk 2]
```diff
@@ -195,3 +204,8 @@ def test_count():
 def test_segment_radial_dists():
     _check_segment_radial_dists(SIMPLE_NEURON)
     _check_segment_radial_dists(SIMPLE_TREE)
+
+
+def test_segment_taper_rate():
```
Comment (mgeplf) [inline neurom/tests/test_segments.py]: taper wasn't added in this diff, right?

File: neurom/segments.py [hunk 3]
```diff
@@ -75,11 +78,16 @@ def radial_dist(pos):
 
     Radial distance calculater WRT pos
     '''
-    return functools.partial(mm.segment_radial_dist, pos=pos)
+    @segment_function(as_tree=False)
```
Comment (mgeplf) [inline neurom/segments.py]: I probably would use functools.partial to bind pos

**Comment by mgeplf** [comment]
seems reasonable to me

File: neurom/segments.py (hunk 3 above)
Comment (juanchopanza) [inline neurom/segments.py]: I tried, and it clashed with my magic stuff. Maybe I was doing it wrong.

File: neurom/tests/test_segments.py (hunk 2 above)
Comment (juanchopanza) [inline neurom/tests/test_segments.py]: I added `taper` to `neurom.segments`.

Comment (mgeplf) [inline neurom/tests/test_segments.py]: ah, right, sorry

**Pull request merged**